File: Solutions/2108. Find First Palindromic String in the Array/2108.py
class Solution:
    def firstPalindrome(self, words: List[str]) -> str:
        def checkPalindrome(word):
            start, end = 0, len(word) - 1
            while start < end:
                if word[start] != word[end]:
                    return False
                start += 1
                end -= 1
            return True
        for word in words:
            if checkPalindrome(word):
                return word
        return ""

File: Solutions/2108. Find First Palindromic String in the Array/2108.java
class Solution {
    public String firstPalindrome(String[] words) {
        for (String word: words) {
            if (checkPalindrome(word)){
                return word;
            }
        }
        return "";
    }
    public boolean checkPalindrome(String word) {
        char[] text = word.toCharArray();
        int start = 0, end = word.length() - 1;
        while(start < end){
            if (text[start] != text[end]){
                return false;
            }
            start ++;
            end --;
        }
        return true;
    }
}